### ISSUE TITLE
Issue With Date Facets (from 1032)

### ISSUE THREAD

**Issue by gigamorph** [project-lux/lux-marklogic#37]
**Problem Description:**  In some situations (unknown as to the scenarios in which it's not working), the date facets when applied generate zero results.

**Example:** A search that includes this record https://lux.collections.yale.edu/view/activity/9ed8ad50-f405-4cf9-9bb7-180048eeb078 has a date of 2017-2017. Applying that generates a search that has no results.

**Expected Behavior:** When applying a facet, there should always be results. In this case it should include the referenced record, rather than no results.  

Link: Note that clicking back from this state results in the facets not being updated. This is[ lux-web#1325 ](https://git.yale.edu/lux-its/lux-web/issues/1325)(closed)

****Dependency/Blocked**
- Dependency of https://github.com/project-lux/lux-frontend/issues/54
- Dependency of https://github.com/project-lux/lux-frontend/issues/76
- Blocked by - https://github.com/project-lux/lux-marklogic/issues/47; https://github.com/project-lux/data-pipeline/issues/122

**Link:** https://lux.collections.yale.edu/view/results/events?q=%7B%22name%22%3A%22small-great%22%7D

**Reference:**
  https://git.yale.edu/lux-its/marklogic/issues/471#issuecomment-14626 included diagrams and modelling
Old tix: https://git.yale.edu/lux-its/marklogic/issues/1032

**Screenshot:** 
![image](https://github.com/project-lux/lux-marklogic/assets/160149818/ae0dc52e-cc42-421f-bf68-6eecfca5129d)

Reference: https://git.yale.edu/lux-its/marklogic/issues/471#issuecomment-14626 included diagrams and modelling

**Links:**
- https://www.bugherd.com/projects/284041/tasks/1616
- https://www.bugherd.com/projects/284041/tasks/1755 
- https://www.bugherd.com/projects/284041/tasks/1746
- https://www.bugherd.com/projects/284041/tasks/1698
- https://www.bugherd.com/projects/284041/tasks/2324
- https://lux-front-tst.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22AND%22%3A%5B%7B%22producedDate%22%3A%221798%22%2C%22_comp%22%3A%22%3E%3D%22%7D%2C%7B%22producedDate%22%3A%221820%22%2C%22_comp%22%3A%22%3C%3D%22%7D%2C%7B%22classification%22%3A%7B%22name%22%3A%22stoneware%22%7D%7D%5D%7D%5D%7D

**Comment by ar2674** [comment]
From old tix: 
![image](https://github.com/project-lux/lux-marklogic/assets/160149818/49c1c576-9108-4255-8b2a-922b9571cae8)

Example Queries: 
[Top left red bar](https://lux-front-dev.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22producedBy%22%3A%7B%22name%22%3A%22Georgia+O%27Keeffe%22%7D%7D%2C%7B%22classification%22%3A%7B%22name%22%3A%22watercolors%22%7D%7D%2C%7B%22name%22%3A%22Flower%22%7D%2C%7B%22producedDate%22%3A%7B%22start%22%3A%221800%22%2C%22end%22%3A%221801%22%7D%7D%5D%7D)
[Top right red bar](https://lux-front-dev.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22producedBy%22%3A%7B%22name%22%3A%22Georgia+O%27Keeffe%22%7D%7D%2C%7B%22classification%22%3A%7B%22name%22%3A%22watercolors%22%7D%7D%2C%7B%22name%22%3A%22Flower%22%7D%2C%7B%22producedDate%22%3A%7B%22start%22%3A%222000%22%2C%22end%22%3A%222001%22%7D%7D%5D%7D)
[Top left green bar](https://lux-front-dev.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22producedBy%22%3A%7B%22name%22%3A%22Georgia+O%27Keeffe%22%7D%7D%2C%7B%22classification%22%3A%7B%22name%22%3A%22watercolors%22%7D%7D%2C%7B%22name%22%3A%22Flower%22%7D%2C%7B%22producedDate%22%3A%7B%22start%22%3A%221880%22%2C%22end%22%3A%221890%22%7D%7D%5D%7D)
[Top right green bar](https://lux-front-dev.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22producedBy%22%3A%7B%22name%22%3A%22Georgia+O%27Keeffe%22%7D%7D%2C%7B%22classification%22%3A%7B%22name%22%3A%22watercolors%22%7D%7D%2C%7B%22name%22%3A%22Flower%22%7D%2C%7B%22producedDate%22%3A%7B%22start%22%3A%221985%22%2C%22end%22%3A%221987%22%7D%7D%5D%7D)
[Middle green bar](https://lux-front-dev.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22producedBy%22%3A%7B%22name%22%3A%22Georgia+O%27Keeffe%22%7D%7D%2C%7B%22classification%22%3A%7B%22name%22%3A%22watercolors%22%7D%7D%2C%7B%22name%22%3A%22Flower%22%7D%2C%7B%22producedDate%22%3A%7B%22start%22%3A%221800%22%2C%22end%22%3A%222000%22%7D%7D%5D%7D)
[Bottom green bar](https://lux-front-dev.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22producedBy%22%3A%7B%22name%22%3A%22Georgia+O%27Keeffe%22%7D%7D%2C%7B%22classification%22%3A%7B%22name%22%3A%22watercolors%22%7D%7D%2C%7B%22name%22%3A%22Flower%22%7D%2C%7B%22producedDate%22%3A%7B%22start%22%3A%221890%22%2C%22end%22%3A%221891%22%7D%7D%5D%7D)

**Comment by brent-hartwig** [comment]
> **Expected Behavior:** When applying a facet, there should always be results. In this case it should include the referenced record, rather than no results.

@jffcamp and @prowns, the expected behavior is not always achievable at scale.  Facets are calculated on unfiltered results --results determined via indexes alone.  We can investigate specific instances but need to reset expectations.

Investigating specific instances may surface a bug or means to index the data better.

**Comment by brent-hartwig** [comment]
The cited record is an event that started in Feb 2017 and ended in Jun 2017:

![image](https://github.com/project-lux/lux-marklogic/assets/13734146/4f39437c-3229-4829-8164-28107062df90)

I am able to reproduce the issue by performing a Simple Search of events matching "Small-Great Objects: Anni and Josef Albers in the Americas, Yale University Art Gallery, New Haven" then using the date facet.

When I repeat as an Advanced Search, I am able to get the search result when start date is `>=` 2017 and/or end date is `<=` 2017.  I cannot get the search result using the other operators, even ones that should.  For instance, start date `>=` 2000 returns the 2017 result but start date `>` 2017 does not (yet should).

I suspect a bug in the data search pattern.

**Comment by brent-hartwig** [comment]
@azaroth42 and @kkdavis14, for the example in the description, we have a disconnect between the query and data:

![image](https://github.com/project-lux/lux-marklogic/assets/13734146/7f4dd7db-a6d7-422e-b761-cba91409059d)

Frontend and backend code are also in play:

* The frontend adds `{"startDate":"2017","_comp":">="}, {"startDate":"2017","_comp":"<="}` when the facet value is selected.
* The backend converts that criteria to:
![image](https://github.com/project-lux/lux-marklogic/assets/13734146/b5d00b72-9018-44d3-bcc5-ff6b57ca9c6c)

It's not clear to me what should change.  What comes to your mind?

Record: https://lux.collections.yale.edu/data/activity/9ed8ad50-f405-4cf9-9bb7-180048eeb078

**Comment by kkdavis14** [comment]
I think it should be does `eventInitiatedStartDateFloat`/`seconds_since_epoch_begin_of_the_begin` fall within 2017 (in epoch time) (or whatever two dates are entered (in epoch time)).

the query is about the start, and the second date we're using here is the end.
with that said, perhaps we need to add a query for end date to events faceting.

**Comment by brent-hartwig** [comment]
@kkdavis14, LUX presently supports a single date-related search pattern.  It is geared towards utilizing the start and end of a timespan where the operator is the deciding factor.  This aligns with @azaroth42's argument/requirement that we should always deal in timespan ranges versus a single point in time.  That said, I believe:

1. It is difficult for users to understand how the backend is applying their date criteria.
2. The frontend may need to change how it specifies date criteria/facets.
3. Whether it be via backend code changes and/or data changes, we may need to close some disconnects.

A meeting may be the best way to proceed --after I have time to investigate additional date-related issue reports.  My next comment will speak to another instance of excluding an item due to operating against two points in time vs. one.

**Comment by brent-hartwig** [comment]
@kkdavis14, you shared a search with me whereby the results were missing a result you expected.  I modified the search to reduce the number of search results but it would include your search result had it not failed the end date portion of the search criteria.

The search is for [objects containing "coffeepot" in the (primary) name that were created between 1730 and 1800](https://lux.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22producedDate%22%3A%221730%22%2C%22_comp%22%3A%22%3E%3D%22%7D%2C%7B%22producedDate%22%3A%221800%22%2C%22_comp%22%3A%22%3C%3D%22%7D%2C%7B%22name%22%3A%22coffeepot%22%7D%5D%7D).  This yields 11 results and excludes https://lux.collections.yale.edu/data/object/000de659-fce8-4b8d-9b4a-c9bb3c97bc61, which was created _sometime between_ 1790 and 1840.  Because this object could have been created between 1790 and 1800, I believe this object should have been included in the search results.  Unlike the previous example, the data has values for both indexes the query used.  However, it required the end of the range to be 1800 or earlier, which would exclude this record as the end of its range is 1840.  Something appears amiss with the "overlapping" portion of the overlapping date range search pattern.  I will investigate and potentially follow-up via separate ticket given there is enough difference between the two examples.

**Comment by brent-hartwig** [comment]
This is a fun one.  Below is my current understanding, some which may contradict previous statements.

- The original overlapping date range search pattern was implemented in [old #471](https://git.yale.edu/lux-its/marklogic/issues/471) / [PR #517](https://git.yale.edu/lux-its/marklogic/pull/517).
- [Old #696](https://git.yale.edu/lux-its/marklogic/issues/696) reported the need for additional date search terms to resolve against the beginning and ending of a timespan, including `agentStartDate`.  This is confusing as there was only one date search pattern after [old #471](https://git.yale.edu/lux-its/marklogic/issues/471).
- [Old #696](https://git.yale.edu/lux-its/marklogic/issues/696) / [PR #744](https://git.yale.edu/lux-its/marklogic/pull/744) _replaced_ the pattern to address the ticket --yet appears to have invalidated some of what [old #471](https://git.yale.edu/lux-its/marklogic/issues/471) could do, specifically constructing a cts.andQuery containing cts.fieldRangeQuery calls with different start and end dates.
- After reviewing [old #471](https://git.yale.edu/lux-its/marklogic/issues/471)'s original implementation today, I believe there are two two bugs in the logic and now wonder if fixing those was all we needed to do.  The suspected bugs were present when [old #696](https://git.yale.edu/lux-its/marklogic/issues/696) replaced the pattern.
- After changing the logic, I tested [Rob's above scenarios](https://github.com/project-lux/lux-marklogic/issues/37#issuecomment-1984010758) in Query Console.  They all worked as expected.
- Further bolstering this direction, there was an attempt to address this ticket when it was still [old #1032](https://git.yale.edu/lux-its/marklogic/issues/1032).  Most of [old #471](https://git.yale.edu/lux-its/marklogic/issues/471)'s implementation was making its way back in.  There's an [extensive Teams discussion](https://teams.microsoft.com/l/message/19:GmLeQu1cBR40mfrno-l8lngg7RFCcJs8fvFIw-MJjIM1@thread.tacv2/1704903709680?tenantId=dd8cbebb-2139-4df8-b411-4e3e87abeb5c&groupId=4620eac6-0955-4c69-93e8-b493b21f0df6&parentMessageId=1704903709680&teamName=MarkLogic&channelName=General&createdTime=1704903709680) on it.  Ultimately, the implementation was backed out until an increased(?) difference between unfiltered and filtered counts could be investigated.  [These backed-out lines](https://git.yale.edu/lux-its/marklogic/commit/bed75983817975fba8b050dd55caa2088c33d23f#diff-278e3a71bc37c3389fadb63989c032bfef7539825f6a5a808cd1a2f1051000d1L233-L246) look most like [old #471](https://git.yale.edu/lux-its/marklogic/issues/471).
- In case someone out there would like this to be a touch more challenging, a gap was identified in [old #696](https://git.yale.edu/lux-its/marklogic/issues/696).  The team elected to address it in [old #759](https://git.yale.edu/lux-its/marklogic/issues/759) / [PR #893](https://git.yale.edu/lux-its/marklogic/pull/893), thereby adding another bit to make sure works as required before declaring victory.
- I also tested the above [Objects that could have been created in 2023 scenario](https://github.com/project-lux/lux-marklogic/issues/37#issuecomment-2007092351):
    - The screenshot shows 10 results and could be missing some.
    - [old #471](https://git.yale.edu/lux-its/marklogic/issues/471)'s implementation with logic errors returned 24 unfiltered / 24 filtered results.
    - [old #471](https://git.yale.edu/lux-its/marklogic/issues/471)'s implementation without logic errors returned **72 unfiltered** / 24 filtered results --perhaps there's something to that pattern that increases false positives.
- Results for the above ["coffeepot" search](https://github.com/project-lux/lux-marklogic/issues/37#issuecomment-2008152180):
    - [old #471](https://git.yale.edu/lux-its/marklogic/issues/471)'s implementation with logic errors returned 11 unfiltered / 11 filtered results.
    - [old #471](https://git.yale.edu/lux-its/marklogic/issues/471)'s implementation without logic errors returned 13 unfiltered / 13 filtered results --so it doesn't always have false positives.  The two that the modified pattern picks up are:
        - `https://lux.collections.yale.edu/data/object/000de659-fce8-4b8d-9b4a-c9bb3c97bc61`
        - `https://lux.collections.yale.edu/data/object/92a6e57e-b0bb-4429-927f-e49fba870475`
- WRT the above [facet issue](https://github.com/project-lux/lux-marklogic/issues/37#issuecomment-2007966015), the first bit we need to determine is whether each Event record should have four points in time within `/json[type= ('Activity', 'Period')]/timespan`.

**Comment by brent-hartwig** [comment]
The following is informed by conversations with and input from Rob, Kelly, and Peter.

## High-Level Conclusions

1. Versions of the date search pattern are yet to correctly implement open- _and_ closed-ended ranges at the same time.
2. The current implementation fully supports open-ended ranges but only partially supports closed-ended ranges.  In the following diagram, only Bar E is supported for closed-ended ranges.

![image](https://github.com/project-lux/lux-marklogic/assets/13734146/422de1c6-2d71-4c45-98be-5c69c8f4f9e0)

3. There is a conflict between the dataset and search whereby the dataset expects search to apply default values to timelines (or otherwise include documents) yet search isn't geared to do so, resulting in scenarios when anticipated documents are excluded from search results.

## Proposed Changes

### Data

- For events, within `/json[type= ('Activity', 'Period')]/timespan/`, collapse (and rename?) the following property pairs to support a single timespan:
    - `begin_of_the_begin` and `_seconds_since_epoch_begin_of_the_begin`
    - `begin_of_the_end` and `_seconds_since_epoch_begin_of_the_end`
    - `end_of_the_begin` and `_seconds_since_epoch_end_of_the_begin`
    - `end_of_the_end` and `_seconds_since_epoch_end_of_the_end`
- For any timespan that is a point in time, set the begin and end properties to the same value.
- For any timespan that is a range and does not have a begin value, set the begin property to the oldest date in LUX (xs:dateTime and float values for each).  **Frontend implication.**
- For any timespan that is a range and does not have an end value, set the end property to the most future date in LUX (xs:dateTime and float values for each).  E.g., ongoing exhibition.  **Frontend implication.**

### Backend

- Update the following indexes to support a single timespan:
    - `eventInitiatedStartDateStr`
    - `eventInitiatedStartDateFloat`
    - `eventInitiatedEndDateFloat`
    - `eventCompletedStartDateStr`
    - `eventCompletedStartDateFloat`
    - `eventCompletedEndDateFloat`
- Delete the `eventEndDate` facet configuration and update the `eventStartDate`’s index name to match that of the new and only event date string index.
- Update the event date search term’s event date float index names.
- The current date search pattern fully supports open-ended ranges but only partially supports close-ended ranges.  The pattern is to be corrected.  **Frontend implication.**

### Middle Tier

TBD

### Frontend

- Detect and strip artificial start and end dates.
- For date search terms and facets:
    - For open-ended ranges, continue specifying the known year with a single search term.
    - For close-ended ranges, begin specifying the known years with a single search term.
        - Delimit the years with a semi-colon.
        - Use the `=` operator for overlapping.
        - Use `!=` for not overlapping.
        - Only one date is required for all other operators.

Example of requiring items encountered between 1600 and 1610, inclusive:

```
{
   "_scope":"item",
   "_comp":"=",
   "encounteredDate":"1600;1610"
}
```

## Next Steps

Before pursing all of the above changes, we should test the direction we're heading:

1. Collect/document a complete set of test cases.
2. Update the search pattern.
3. Test via query console or backend endpoint.
    - Verify results are accurate.
    - Document filtered and unfiltered counts.  May need to explore options if pattern changes result in a greater difference between the two.
    - Skip Events.  All other record types should be fair game.

**Comment by brent-hartwig** [comment]
[Teams thread](https://teams.microsoft.com/l/message/19:GmLeQu1cBR40mfrno-l8lngg7RFCcJs8fvFIw-MJjIM1@thread.tacv2/1715619809928?tenantId=dd8cbebb-2139-4df8-b411-4e3e87abeb5c&groupId=4620eac6-0955-4c69-93e8-b493b21f0df6&parentMessageId=1715619809928&teamName=MarkLogic%20%2F%20Cultural%20Heritage%20Support&channelName=General&createdTime=1715619809928) checking in with Rob and Kelly on the above-proposed timespan data changes.

**Comment by roamye** [comment]
From the teams thread link above, an issue was listed: https://github.com/project-lux/data-pipeline/issues/22. This issue is now closed. Is there anything else needed to move this forward with the proposed changes listed above?

cc: @prowns

**Comment by brent-hartwig** [comment]
> Is there anything else needed to move this forward with the proposed changes listed above?

@roamye, if inclusive of the Next Steps suggested at the bottom of the [same comment](https://github.com/project-lux/lux-marklogic/issues/37#issuecomment-2030174146) the Proposed Changes are in, then no.  But I would advise against just diving into this ticket's implementation.  I think we need a means to comprehensively address issues with date searches and be able to use that means in the future for regression testing.  I'm open to what that means is, including bringing back automated unit tests.

**Comment by brent-hartwig** [comment]
@roamye, do you know why this is still marked as blocked?  I see the description states it is blocked by #47, but I whole-heartedly disagree that this ticket should be held up by that one.  I'm also pretty sure YPM will need the ability to correctly search by dates.

cc: @prowns, @jffcamp

**Comment by brent-hartwig** [comment]
@roamye, my [April Fool's Day analysis](https://github.com/project-lux/lux-marklogic/issues/37#issuecomment-2030174146) factored the cited Teams conversation in and surfaced multiple issues with the current implementation.  I believe we need to resolve some of those issues and then attempt to validate or invalidate estimate issues (#47).

**Comment by roamye** [comment]
@brent-hartwig - 
ahh okay, in that case #47 shouldn't block this one.

any thoughts? @prowns / @jffcamp

**Comment by prowns** [comment]
@clarkepeterf  - this is listed as "in progress" - should it be attached to 9/9 milestone?

**Comment by clarkepeterf** [comment]
@prowns No, I'm not actively working on this. But we should discuss in the team meeting tomorrow

**Comment by clarkepeterf** [comment]
@kkdavis14 @azaroth42 while https://github.com/project-lux/data-pipeline/issues/22 fixes some issues with dates, it does not fix searching for Events because they also rely on `begin_of_the_end`, `_seconds_since_epoch_begin_of_the_end`,
`end_of_the_begin` and `_seconds_since_epoch_end_of_the_begin`

**Comment by azaroth42** [comment]
Can ML indexes have precedence for values? e.g. use begin_of_the_end if it exists otherwise use end_of_the_end ?

**Comment by brent-hartwig** [comment]
I dug into that question a while ago and will need to dig up my notes.  Added to my TODO list.

**Comment by clarkepeterf** [comment]
@azaroth42 @brent-hartwig There isn't a way to do a straight-up fallback value in ML indexes. We can have multiple paths with different weights. For example: `eventCompletedStartDate` could match `begin_of_the_end` with weight 2, and `end_of_the_end` with weight 1. So if a document it has both, begin_of_the_end would rank higher. But it could also potentially match documents where `end_of_the_end` is in the range, but `begin_of_the_end` is not

**Comment by azaroth42** [comment]
Is this only for the timespan on Event records? Or for all timespans everywhere?

If it's only the one record type, and thus only the one path ... what about magic at the javascript layer?

**Comment by brent-hartwig** [comment]
My recollection is that we only knew this to impact events but that if we encountered another record type with the same issue, the same decision of setting all since-epoch props was to apply.  I'm open to revisiting means for the backend to address this.

**Comment by clarkepeterf** [comment]
@azaroth42  At present, only event search terms use `begin_of_the_end` or `end_of_the_begin`, but if future fields use `begin_of_the_end` or `end_of_the_begin`, a including them would also help those terms


My initial thought for a background workaround was the code below, but it basically ends up just matching if either `begin_of_the_end` or `end_of_the_end` are in the range. This would give false positives where `begin_of_the_end` is not in the range but `end_of_the_end` is in the range. I think the issue is that when we get results we don't know if a record has been excluded because:
1.  its value is not in the range
or
2. that property is not in the data

example: `eventCompletedStartDate > 2010`
```
cts.orQuery([
	cts.fieldRangeQuery(${begin_of_the_end_field}, '>', ${2010 in seconds})
	cts.andQuery([
		cts.fieldRangeQuery(${end_of_the_end_field}, '>', ${2010 in seconds})
		cts.notQuery([
			cts.fieldRangeQuery(${begin_of_the_end_field}, '>', ${2010 in seconds})
		])
	])
	
])
```

**Comment by roamye** [comment]
From issue #43 : 
@brent-hartwig 
### This Ticket

> Facets only returns start dates.

Very true.  We could address this by:

1. Identify the end dates and ensure their string values are indexed.
2. Modify facetConfig.mjs to support multiple indexes per facet, whether as a single or second property.  In other contexts, we renamed the `indexReference` string property to the `indexReferences` string array property yet the next item may encourage we go with a second property instead.
3. If necessary, update generateRemainingSearchTerms.mjs' use of facetConfig.mjs.
4. Update facetLib.mjs' call to cts.fieldValues to use all indexes associated with a facet.

### Optimization

_Whether as part of this ticket's scope or another..._

Thinking about the above has led to an optimization idea as well.  If the frontend only needs the first and last dates, the facets endpoint could offer a parameter saying as much, yet not be limited to dates.  The following illustrates what the implementation could look like where `q` would be the search query.

```
const q = cts.documentQuery([
  'https://lux.collections.yale.edu/data/person/098de228-41f2-404e-83e2-e48b2bd632f8'
]);
const fields = ['agentBornStartDateStr', 'agentDiedStartDateStr'];
const baseOptions = ['limit=1', 'score-zero', 'lazy'];
const results = {
  earliest: cts.fieldValues(fields, null, baseOptions.concat(['ascending']), q),
  latest: cts.fieldValues(fields, null, baseOptions.concat(['descending']), q),
};
results;
```

If we didn't want the endpoint consumer to control this behavior, we could extend the facet configuration.  My preference is a new endpoint parameter that defaults to returning the first and last values for date facets.  Here is how the remaining search terms generator already identifies date facets: `const isDate = facetIndexReference.endsWith('DateStr')`.  If we add a parameter, it should be documented in the API usage doc.

I would expect the facet to be calculated faster and, for some searches, a _much_ smaller response size.

cc: @clarkepeterf | [Link to comment in original issue](https://github.com/project-lux/lux-marklogic/issues/43#issuecomment-2037534354)

-----------------------------------------------
@clarkepeterf, please review the above suggestion in the context of https://github.com/project-lux/lux-marklogic/issues/256. [Link to comment in original issue. ](https://github.com/project-lux/lux-marklogic/issues/43#issuecomment-2275884726)

**Comment by roamye** [comment]
Moved this to backlog and removed blocked label as: https://github.com/project-lux/data-pipeline/issues/122 is closed and no longer blocking this ticket.

**Comment by clarkepeterf** [comment]
I believe the initial problem described in this ticket is fixed. I can apply date facets for this search, and the results still show up: https://lux-front-tst.collections.yale.edu/view/results/events?q=%7B%22AND%22%3A%5B%7B%22text%22%3A%22small-great%22%2C%22_lang%22%3A%22en%22%7D%5D%7D

**Comment by brent-hartwig** [comment]
I used this ticket as a hub for multiple date issues.   It may include information helpful for remaining issues but shouldn't be kept open if the ticket's scope is fixed.

**Comment by clarkepeterf** [comment]
This can be closed - no longer needed after Optic changes
